### DISCUSSION TITLE
can export static page without react ?

### DISCUSSION THREAD

**Discussion by zerofront** [vercel/next.js#38366] [Help]
If I don't need to dynamically change the content of the page, but only need a pure static page, why does the page after `next build && next export` still include the code of react and next?

<img width="715" alt="image" src="https://user-images.githubusercontent.com/19327366/177506404-2ad3a5f1-e308-4c2f-a428-11d9f344ac3e.png">
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/19327366/177506287-b8d4caf8-13af-470c-9931-6234e85dec2d.png">

**Comment by icyJoseph** [comment]
A few things:

- You are using `JSX`, and `JSX` is transformed to `React.createElement` calls, or even more optimal transformations, starting with React 17. See here, https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-different-in-the-new-transform. The concept of JSX pragma is worth exploring, https://www.gatsbyjs.com/blog/2019-08-02-what-is-jsx-pragma/.

- Next.js is still using React though, for SPA routing, and other stuff. It wraps the `non-react` page with components that use `React` state management.

One way out would be to rewrite `/no-react` to an `api/route` and from there send text with `<div>Hello, I am very static</div>`, but that'll miss out on stylesheets, and any features implemented in `_app` and `_document`.

I do think this is could be a bit of a pitfall if the wrong assumptions are made. For example you also can't really edit that `export` output in a sensible way anyway, because **hydration** will find mismatching UI, and at best just override your changes to the static HTML.

If you see the `no-react` JS file, you'll see the transformed JSX of this page! 

One last thing I forgot to mention is that, with JS disabled, the export works well for the most part. Any interactivity after Time To Visible, doesn't work though.

**Reply by zerofront** [reply]
I do find jsx function in no-react.js, but this is also the point of my doubts, this page does not actually need hydration

**Reply by icyJoseph** [reply]
Yet it is hydrated. This is one selling point of other solutions. Something like Astro for example, tries to not send JS unless there's interactivity.

All in all, this is a sort of pitfall of SSR. The Time To Visible is incredibly short, but then hydration kicks in, so Time To Interactivity is still a thing to consider. On the upside, if JS is disabled, Time To Visible still shows the server rendered output and we are good to go!

And I know, your user code (the page) is pretty simple, but there's a couple of Next.js internals that wrap that page, with React stuff. 

Is the question whether or not the framework is in condition of statically analyzing each page and concluding whether or not the framework supporting code needs to be downloaded at high priority, or later, or on demand? Then, at the moment, no, not yet.

**Reply by zerofront** [reply]
thanks, i think i get it
